lib/kcerts/ssh.go

```diff
@@ -100,6 +106,51 @@ func (a SSHAgent) Valid() bool {
 	return err == nil
 }
 
+func (a *SSHAgent) UseStandardPaths() error {
+	user, err := UserCurrent()
+	if err != nil {
+		return err
+	}
+
+	// Make sure /tmp/ssh-${USER} exists with the right permissions
+	path := fmt.Sprintf("/tmp/ssh-%s", user.Username)
```

Comment:
This code makes me very uncomfortable, I don't think it's safe to create a file with a predictable name in /tmp, ever (although I see checks below that may prevent most problems), and code below looks racy. Eg, another user could create /tmp/ssh-jonathan, socket may be created in between the stat and remove, ...

Reply:
(why use /tmp, and not a dir created by the enkit cache, that is shared across containers?)

(also, enkit agent saves the agent information in the cache already, whatever the path is, it can re-use it, do a simple cd ~/.cache/enkit; find . -type f -name ssh.json)

Reply:
(also, see comment below on /tmp, I would not use hard links)

Reply:
You should be uncomfortable!  This is security-critical code and many of our forebears have been crushed in these trenches.

1. Is it safe to create a directory with a predictable name / another user can create /tmp/ssh-jonathan?

Only if we can guarantee that the user has exclusive access to that directory, and that an attacker can't replace that directory with a fs node they can modify.  This process will fail if we cannot chmod /tmp/ssh-jonathan to 0700, but I should probably also add a check that the directory ends up with the right ownership and mode.

2. Race between stat and remove?

I don't think this is an issue as the directory is owned by the user and chmod 0700 at this point -- so nobody but the user can add or remove nodes in that directory at this point.  Failure to remove an existing socket node results errors out.

3. Why use /tmp?

Because sshd is already creating /tmp/ssh-XXXXX for the forwarded agent socket, and I wanted to keep the new name in the same directory / on the same filesystem.  If I switch to using a symlink, this is less of a requirement.

Reply:
Done.